### PR TITLE
Add a TUF API to aklite

### PR DESCRIPTION
**Pull request by detsch**
The API allows a better isolation between aktualizr-lite and libaktualizr, and also allows the use of multiple sources for a same TUF repository.

Changes were originally submitted in #279

### CODE REVIEW

**Comment by mike-sul** [comment]
LGTM. Let's bitbake and test it e2e (focus on the custom client, make sure no regression there).

**Review by mike-sul — APPROVED** [review]
I think this is ready to go.

File: apps/tufctl/main.cpp [hunk 1]
```diff
@@ -0,0 +1,106 @@
+#include <boost/property_tree/ini_parser.hpp>
```
Comment (doanac) [inline apps/tufctl/main.cpp:1]: This seems like its more for debug/development? I'm not sure what the long term point of this is - is more of an "examples"?

Comment (mike-sul) [reply apps/tufctl/main.cpp:1]: Yes, this is rather a test and an example utility. That's why it's not built by default.

Comment (doanac) [reply apps/tufctl/main.cpp:1]: y'all put a lot of thought into this so i won't bikeshed. But it seems like we might be developing an "examples" directory over time.

**Review by doanac — APPROVED** [review]

**Comment by mike-sul** [comment]
@detsch Feel free to merge it.

**Comment by detsch** [comment]
I've done some tests yesterday using imx8mm with SE05X, focusing on custom-client. All seems OK.

**Comment by mike-sul** [comment]
> I've done some tests yesterday using imx8mm with SE05X, focusing on custom-client. All seems OK.

Let's merge it then.

**Pull request merged**